contracts/arbitrum/connectors/spell/main.sol

```diff
@@ -0,0 +1,77 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @title DSA Spell.
+ * @dev Cast spells on DSA.
+ */
+
+// import files
+import { AccountInterface } from "../../common/interfaces.sol";
+import { Stores } from "../../common/stores.sol";
+import { Events } from "./events.sol";
+
+abstract contract DSASpell is Events, Stores {
+	/**
+	 *@dev Cast spells on DSA.
+	 *@param targetDSA target DSA to cast spells on.
+	 *@param connectors Array of connector names.
+	 *@param datas Array of connector calldatas.
+	 */
+	function castDSA(
+		address targetDSA,
+		string[] memory connectors,
+		bytes[] memory datas
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		require(instaList.accountID(targetDSA) != 0, "not-a-DSA");
+
+		AccountInterface(targetDSA).cast(connectors, datas, address(this));
+
+		_eventName = "LogCastDSA(address,string[],bytes[])";
+		_eventParam = abi.encode(targetDSA, connectors, datas);
+	}
+
+	/**
+	 *@dev Perform spells.
+	 *@param connectors Array of connector names.
+	 *@param datas Array of connector calldatas.
+	 */
+	function castSpells(string[] memory connectors, bytes[] memory datas)
```

Comment:
```suggestion
	function retrySpell(string[] memory connectors, bytes[] memory datas)
```

Reply:
@Richa-iitr @pradyuman-verma Any better name suggestion?

Reply:
Ahh, I think `spellFactory` will be better.